### DISCUSSION TITLE
2FA for web extension

### DISCUSSION THREAD

**Discussion by Unlisted1723** [dani-garcia/vaultwarden#4746] [Ideas]
Hello,

I was wondering if it would be possible to require 2FA every time you want to unlock your vault with the web extension. Currently, you are asked for 2FA only on the first connection, but never after. So what is the purpose of having 2FA if you use it only once?

I noticed that you can set the session timeout to disconnect your account. However, this option works only if the user intentionally uses it and agrees to re-enter their email. Unfortunately, many of my colleagues will never do so.

So my question is: will there be an option to use 2FA when unlocking your vault? Or an option in the admin panel to force the logout after session timeout?

Best regards.

**Comment by RealOrangeOne** [comment]
This is something which would need implementing upstream in the Bitwarden extension itself. You'd be better raising an issue there.

**Reply by Unlisted1723** [reply]
Thank you for the info.
